woocommerce/woocommerce-gutenberg-products-block
# Add line pricing for cart items in the API

#1963 is caused by rounding; the API returns rounded prices. Hard to explain but to summarise:

- Taxes are calculated per line, not per item
- If you round item prices first before multiplying by qty, you end up with a higher amount. 
- We need to use unrounded values, or return line prices via the API 

In the example @Aljullu gave in #1963 the cart page was doing this:

14.55 * 5 = 72.75

The cart page (core) is doing this:

14.54545500 * 5 = 72.727 (rounds to 72.73)

Hence the mismatch. This PR solves it by returning line_price properties with the qty factored in already.

Fixes #1963

### How to test the changes in this Pull Request:

1. Input prices inc tax. 
2. Setup product costing 16.
3. Setup tax rate of 10%
4. Add 5 of those items to cart.
5. Subtotal and line total should match.
